### ISSUE TITLE
Ref value is not accurate in nested mount

### ISSUE THREAD

**Issue by chronoDave** [forgojs/forgo#62]
I was getting strange results from calling `getBoundingClientRect()` on a ref value after mounting, the element's height was always 0. However, when logging the value inside a timeout, it seems to return the actual height. Stranger still is this only happens in nested components.

**Nested mount**
![image](https://user-images.githubusercontent.com/27073716/171385032-3d69744d-cb52-41be-b82a-0e13d8d33c84.png)

```tsx
import { createElement, mount } from 'forgo';

const List = () => {
  let ref = {};

  return {
    mount: () => {
      console.log(ref.value.getBoundingClientRect()); // 0
      setTimeout(() => console.log(ref.value.getBoundingClientRect()), 0); // 927
    },
    render: () => (
      <div style="overflow: auto; flex-grow: 1" ref={ref}>
        <div style="position: relative; min-height: 0">
          <div style="position: absolute">Hello world</div>
        </div>
      </div>
    )
  }
};

const App = () => {
  return {
    render: () => (
      <div style="display: flex; flex-direction: column; min-height: 100vh">
        <List />
      </div>
    )
  };
};

mount(<App />, document.getElementById('mount'));
```

**"Flat" mount**
![image](https://user-images.githubusercontent.com/27073716/171385947-605336e7-34ec-43e4-add9-2edaa1ae5eb5.png)

```tsx
import { createElement, mount } from 'forgo';

const App = () => {
  let ref = {};

  return {
    mount: () => {
      console.log(ref.value.getBoundingClientRect());
      setTimeout(() => console.log(ref.value.getBoundingClientRect()), 0);
    },
    render: () => (
      <div style="display: flex; flex-direction: column; min-height: 100vh">
        <div style="overflow: auto; flex-grow: 1" ref={ref}>
          <div style="position: relative; min-height: 0">
            <div style="position: absolute">Hello world</div>
          </div>
        </div>
      </div>
    )
  };
};

mount(<App />, document.getElementById('mount'));
```

**Comment by chronoDave** [comment]
I'm not sure, but is it related to #50 ?

**Comment by chronoDave** [comment]
After some debugging I found the following:

![image](https://user-images.githubusercontent.com/27073716/171412205-2c8d6873-ba82-4caa-a381-c99cd15563f3.png)

I can't really make head nor tails of it, but I hope it helps

**Comment by spiffytech** [comment]
@chronoDave We won't be able to merge the fix until it completes code review, but I've published the current PR code in `forgo@3.1.2-beta.0` to tide you over until then.

**Comment by spiffytech** [comment]
This has been released in Forgo v3.2.0.
